source/validate_cfg.cpp

```diff
@@ -284,41 +418,39 @@ spv_result_t PerformCfgChecks(ValidationState_t& _) {
       }
     }
 
-    // Check all referenced blocks are defined within a function
-    if (function.get_undefined_block_count() != 0) {
-      std::stringstream ss;
-      ss << "{";
-      for (auto undefined_block : function.get_undefined_blocks()) {
-        ss << _.getIdName(undefined_block) << " ";
-      }
-      return _.diag(SPV_ERROR_INVALID_CFG)
-             << "Block(s) " << ss.str() << "\b}"
-             << " are referenced but not defined in function "
-             << _.getIdName(function.get_id());
-    }
-
-    // Check all headers dominate their merge blocks
+    // Check construct rules
     for (Construct& construct : function.get_constructs()) {
-      auto header = construct.get_header();
-      auto merge = construct.get_merge();
-      // auto cont = construct.get_continue();
+      auto header = construct.get_entry();
+      auto merge = construct.get_exit();
 
+      //  for a given loop, its back-edge block must post dominate the
+      //  OpLoopMerge's Continue Target , and that Continue Target must dominate
+      //  that back edge block
```

Comment:
The comment here makes it seem like you're only checking loops.  But this first bit's comment should really be: if the merge block is reachable, then it's dominated by the header.